### ISSUE TITLE
Fix SSJ bug where invalid command triggers repeat

### ISSUE THREAD

**Issue by fatcerberus** [spheredev/neosphere#152]
Like GDB, some SSJ commands can be repeated by leaving the command line empty.  However, repeats are also triggered by an *unrecognized* command, which is incorrect behavior and may even screw up a debug session in the case of stepping commands:

```
scripts/main.js:5 eval()
(ssj) list
      1 /***
      2  * Specs Engine v6: Spectacles Saga Game Engine
      3   *           Copyright (c) 2017 Power-Command
      4 ***/
=>    5 var DayNightEngine = require('@/scripts/inGameClock').DayNightEngine;
      6 var Delegate = require('events').Delegate;
      7 var _a = require('scenes'), defScenelet = _a.defScenelet, Scene = _a.Scene;
      8 var Thread = require('threads').Thread;
      9 global.from = require('from');
     10 global.joy = require('joy');

scripts/main.js:5 eval()
(ssj) geteaten
`geteaten`: unrecognized command name.
     10 global.joy = require('joy');
     11 global.music = require('music');
     12 global.prim = require('prim');
     13 global.random = require('random');
     14 global.term = require('term');
     15 global.threads = require('threads');
     16 RequireScript('battleEngine/battle.js');
     17 RequireScript('gameOverScreen.js');
     18 RequireScript('menuStrip.js');
     19 RequireScript('scenelets.js');
```

Last time I attempted to fix this bug, it required some rearchitecting of the command reading code, so I put it off--and then promptly forgot about it.  So this issue is a reminder that it still needs to be fixed.

**Comment by fatcerberus** [comment]
Concretely, the issue is here:
https://github.com/fatcerberus/minisphere/blob/master/src/ssj/session.c#L220

`resolve_command()` returns NULL if the command-line is empty (in which case we *want* to repeat the last command), but also if the command entered is invalid and could not be resolved.  It's not currently possible for the code handling repeats to distinguish between the two cases.

One way to solve the issue would be for `resolve_command()` to return something like "repeat" as the verb for an empty command line, but that seems like a hack to me; I'd prefer a less fragile solution if possible.  I'll give it some thought.

**Comment by fatcerberus** [comment]
This is fixed in 4.5.11.  Rather than relying on the `NULL` return of `resolve_command()` (which, as mentioned, is ambiguous), I just had it check for a zero-length command line directly to decide whether to do a repeat.

Closing the issue.
